matlab/general_least_squares_error.m

```diff
@@ -80,39 +71,30 @@
 
     strain_x1 = [strain_x1;x1];
 
-    nSE = 0;
-    errMatStrain = 0;
-    
-    % This probably will eventually replace errMatStrain -TR
-    errorVal = 0;
-    
-%     for gene = 1:num_genes        
-        for iT = 1:length(expression_timepoints)
-%             Some changes here -TR
-%             truncatedData = log2FC(qq).expressionData(2:end, :);
-%             dataCell = truncatedData{gene, iT};
-% 
-%             for replicate = 1:length(dataCell)
-%                 errorVal = errorVal + (log2(x1(iT, gene)) - dataCell(1, replicate))^2;
-%                 nSE      = nSE + 1;
-%             end
+    for timepointIndex = 1:length(expression_timepoints)        
+        for geneIndex = 1:num_genes
+            truncatedData = log2FC(qq).compressed(2:end, :);
+            dataCell = truncatedData{geneIndex, timepointIndex};
             
-            for iF =  1:length(log2FC(qq).t(iT).indx)
-                errMatStrain = errMatStrain+((log2(x1(iT,:)))'-d(:,log2FC(qq).t(iT).indx(iF))).^2;
-                nSE      = nSE + 1;
+            for replicateIndex = 1:size(dataCell, 2)                
+                errorWholeModel(geneIndex) = errorWholeModel(geneIndex) + ...
+                    (log2(x1(timepointIndex, geneIndex)) - dataCell(1, replicateIndex))^2;
+                nGSE(geneIndex) = nGSE(geneIndex) + 1;
+                nData = nData + 1;
             end
         end
-        errormat = errormat + errMatStrain;
-
-        SSE(:,qq) = errMatStrain/nSE;
-%     end
+        
+        for geneIndex = 1:num_genes
+            MSE(geneIndex, qq) = errorWholeModel(geneIndex) / nGSE(geneIndex);

```

Comment:
This is likely an erroneous calculation because `errorWholeModel` is initialized outside the `qq` loop. Add a `, qq` index to separate the error amounts by strain.

Oh and run that 30-minute MSE test. Actually, run the MSE test first to verify the regression. (but even if there isn't a regression, we are fairly certain that this is incorrect)